clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -1280,42 +1267,37 @@ protected void produceToStoreBufferServiceOrKafka(
-        if (partitionConsumptionState == null) {
-          LOGGER.warn(
-              "Ignoring exception for partition {} for store version {} since this partition has been unsubscribed already.",
-              exceptionPartition,
-              kafkaVersionTopic,
-              exceptionPartitionPair.getException());
-          continue;
-        }
+  private void processIngestionException() {
+    for (int exceptionPartition = 0; exceptionPartition < partitionIngestionExceptionList
+        .size(); ++exceptionPartition) {
+      PartitionExceptionInfo partitionExceptionInfo = partitionIngestionExceptionList.get(exceptionPartition);
+      if (partitionExceptionInfo == null) {
+        continue;
+      }
+      Exception partitionException = partitionExceptionInfo.getException();
+      PartitionConsumptionState partitionConsumptionState = partitionConsumptionStateMap.get(exceptionPartition);
+      if (partitionConsumptionState == null) {
+        LOGGER.warn(
+            "Ignoring exception for partition {} for store version {} since this partition has been unsubscribed already.",
+            exceptionPartition,
+            kafkaVersionTopic,
+            partitionException);
+        continue;
+      }
 
-        if (!partitionConsumptionState.isCompletionReported()) {
-          reportError(
-              exceptionPartitionPair.getException().getMessage(),
-              exceptionPartition,
-              exceptionPartitionPair.getException());
-          if (partitionConsumptionStateMap.containsKey(exceptionPartition)) {
-            unSubscribePartition(kafkaVersionTopic, exceptionPartition);
-          }
-        } else {
-          LOGGER.error(
-              "Ignoring exception for partition {} for store version {} since this partition is already online. "
-                  + "Please engage Venice DEV team immediately.",
-              exceptionPartition,
-              kafkaVersionTopic,
-              exceptionPartitionPair.getException());
-        }
-      } catch (InterruptedException e) {
-        Thread.currentThread().interrupt();
-        throw new VeniceException(
-            "Interrupted while trying to drain exceptions from " + exceptionGroup + " exception queue for topic: "
-                + kafkaVersionTopic);
+      if (!partitionConsumptionState.isCompletionReported()) {
+        reportError(partitionException.getMessage(), exceptionPartition, partitionException);
+      } else {
+        LOGGER.error(
+            "Ignoring exception for partition {} for store version {} since this partition is already online. "
+                + "Please engage Venice DEV team immediately.",
+            exceptionPartition,
+            kafkaVersionTopic,
+            partitionException);
+      }
+      // Unsubscribe the partition to avoid more damages.
+      if (partitionConsumptionStateMap.containsKey(exceptionPartition)) {
+        unSubscribePartition(kafkaVersionTopic, exceptionPartition);
```

Comment:
Curious what is the benefit of unsub ? would it lead to stale data ? what if the error was due to transient issue.

Reply:
If we don't unsub the errored partition, there will be a data gap issue or useless consumption.

For example, with the previous logic, if a hybrid partition encountered an exception:
1. SIT will mark the `errorPartition`, but the main thread will clear it.
2. If the following messages succeed since it was a transient issue, there will be a data gap in Kafka or local RocksDB.
If we don't unsub, with the current logic, there will be more messages coming to the error partition, and drainer will keep ignoring the messages from the same partition, which is a big waste.

The gap is that today, we don't have a good way to expose the current status since we couldn't put the errored hybrid partition into `error` state, so that Controller could reset, and currently, it is only visible via InGraph metrics.
Maybe we could consider to expose this state in offline push status and let Controller scan and reset the error ones.

Reply:
`If we don't unsub, with the current logic, there will be more messages coming to the error partition, and drainer will keep ignoring the messages from the same partition, which is a big waste.`

   But we are going to clear the error partition in main thread. 
 
 If we unsub immediately for any single error for a completed partition, unless we have some metric, it will serve stale data ?

Reply:
There is a metric to catch the stale partitions:
```
partitionIngestionExceptionList.forEach(ep -> {
      if (ep != null && ep.isReplicaCompleted()) {
        versionedIngestionStats.recordStalePartitionsWithoutIngestionTask(storeName, versionNumber);
      }
    });
```
Continuing subscribing the partition is useless.

SIT won't clear the error partitions until the replica is dropped or reseted.

Reply:
The stale partition is expected when encountering such issues, otherwise, there will be data gap issue, which is even more serious.